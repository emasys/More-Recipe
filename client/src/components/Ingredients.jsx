import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const generateList = ({ ingredients }) => {
  if (ingredients) {
    const list = ingredients.recipe.ingredients;
    return list.map((item, index) => (
      <a
        href={`https://en.wikipedia.org/wiki/${item}`}
        target="_blank"
        className="list-group-item text-capitalize list-group-item-action"
        key={index}
      >
        {item}
      </a>
    ));
  }
};

const Ingredients = props => (
  <div className="col-lg-5 col-sm-12">
    <h2 className="fresh-title text-capitalize">
      {props.ingredients ? props.ingredients.recipe.name : 'loading...'}
    </h2>
    <small className="text-capitalize">
      A recipe by{' '}
      <Link
        className=" bolder text-info"
        to={`/user/${props.data ? props.data.data.id : null}`}
      >
        {props.data ? props.data.data.moniker : 'loading...'}
      </Link>
    </small> <br/>
    <small className="">
      Posted on: {props.ingredients ? moment(props.ingredients.recipe.createdAt).format('MMM Do YYYY') : "loading..."}
    </small>
    <hr />
    <h5 className="text-muted">Description</h5>
    <div className="pb-3">
      <div className="bg-light p-15">
        {props.ingredients ?
          props.ingredients.recipe.description :
          'loading...'}
      </div>
    </div>
    <hr />
    <h5 className="text-muted">Ingredients</h5>
    <hr />
    <div className="pb-3">
      <ul className="list-group">{generateList(props)}</ul>
    </div>
    <h5 className="text-muted">Directions</h5>
    <hr />
    <div className="p-10 direction rounded  bg-light">
      {props.ingredients ? props.ingredients.recipe.direction : 'loading...'}
    </div>
  </div>
);

export default Ingredients;
