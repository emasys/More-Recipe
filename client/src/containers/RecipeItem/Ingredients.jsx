import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import PropTypes from 'prop-types';

const generateList = ({ ingredients }) => {
  if (ingredients.recipe) {
    const list = ingredients.recipe.ingredients;
    return list.map((item, index) => (
      <a
        href={`https://en.wikipedia.org/wiki/${item}`}
        target="_blank"
        className="list-group-item text-capitalize list-group-item-action"
        key={index}
      >
        {item}
      </a>
    ));
  }
};

const Ingredients = props => (
  <div className="col-lg-5 col-sm-12">
    <h2 className="fresh-title wrapWord text-capitalize">
      {props.ingredients.recipe ? props.ingredients.recipe.name : 'loading...'}
    </h2>
    <small className="text-capitalize">
      A recipe by{' '}
      <Link
        className=" bolder text-info"
        to={`/user/${props.data ? props.data.data.id : null}`}
      >
        {props.data ? props.data.data.moniker : 'loading...'}
      </Link>
    </small>{' '}
    <br />
    <small className="">
      Posted on:{' '}
      {props.ingredients.recipe ?
        moment(props.ingredients.recipe.createdAt).format('MMM Do YYYY') :
        'loading...'}
    </small>
    <hr />
    <h5 className="text-muted">Description</h5>
    <div className="pb-3 ">
      <div className="bg-light direction rounded p-15">
        {props.ingredients.recipe ?
          props.ingredients.recipe.description :
          'loading...'}
      </div>
    </div>
    <hr />
    <h5 className="text-muted">Ingredients</h5>
    <hr />
    <div className="pb-3">
      <ul className="list-group wrapWord">{generateList(props)}</ul>
    </div>
    <h5 className="text-muted">Directions</h5>
    <hr />
    <div className="p-10 direction rounded  bg-light">
      {props.ingredients.recipe ? props.ingredients.recipe.direction : 'loading...'}
    </div>
  </div>
);

Ingredients.propTypes = {
  ingredients: PropTypes.object,
  data: PropTypes.object
};

export default Ingredients;
