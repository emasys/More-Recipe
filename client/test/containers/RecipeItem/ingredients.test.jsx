import React from 'react';
import { shallow } from 'enzyme';

// Component
import Ingredients from '../../../src/containers/RecipeItem/Ingredients';
// import { AddRecipe } from '../../../src/containers/AddRecipe/index';
import fakeStore from '../../__mocks__/fakeStore';

let wrapper = null;

describe('Test suite for add recipe page', () => {
  const props = {
    ingredients: fakeStore.recipes.recipeItem,
    data: fakeStore.user.userInfo
  };

  beforeEach(() => {
    wrapper = shallow(<Ingredients {...props} />);
  });
  it('Should render without errors', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
