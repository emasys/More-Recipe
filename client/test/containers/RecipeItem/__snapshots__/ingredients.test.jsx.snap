// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for add recipe page should match snapshot 1`] = `
<div
  className="col-lg-5 col-sm-12"
>
  <h2
    className="wrapWord text-capitalize"
  >
    How to cook yam
  </h2>
  <small
    className="text-capitalize"
  >
    A recipe by
     
    <Link
      className=" bolder text-info"
      replace={false}
      to="/user/1"
    >
      admin
    </Link>
  </small>
   
  <br />
  <small
    className=""
  >
    Posted on:
     
    Mar 13th 2018
  </small>
  <hr />
  <h5
    className="text-muted"
  >
    Description
  </h5>
  <div
    className="pb-3 "
  >
    <div
      className="bg-light direction rounded p-15"
    >
      just do it
    </div>
  </div>
  <hr />
  <h5
    className="text-muted"
  >
    Ingredients
  </h5>
  <hr />
  <div
    className="pb-3"
  >
    <ul
      className="list-group wrapWord"
    >
      <a
        className="list-group-item text-capitalize list-group-item-action"
        href="https://en.wikipedia.org/wiki/water, salt"
        key="water, salt"
        target="_blank"
      >
        water, salt
      </a>
    </ul>
  </div>
  <h5
    className="text-muted"
  >
    Directions
  </h5>
  <hr />
  <div
    className="p-10 direction rounded  bg-light"
  >
    just do it
  </div>
</div>
`;
